cluster/cluster.go

```diff
@@ -337,23 +344,55 @@ func (executor *GPDBExecutor) ExecuteLocalCommandWithContext(commandStr string,
 	return string(output), err
 }
 
+// Create a new exec.Command object so we can run it again
+func resetCmd(cmd *exec.Cmd) *exec.Cmd {
+	path := cmd.Path
+	args := cmd.Args
+	return exec.Command(path, args...)
+}
+
+/*
+ * ExecuteClusterCommandWithRetries, but only 1 attempt to keep the previous functionality
+ */
+func (executor *GPDBExecutor) ExecuteClusterCommand(scope Scope, commandList []ShellCommand) *RemoteOutput {
+	return executor.ExecuteClusterCommandWithRetries(scope, commandList, 1, 0)
+}
+
 /*
  * This function just executes all of the commands passed to it in parallel; it
  * doesn't care about the scope of the command except to pass that on to the
  * RemoteOutput after execution.
+ *
+ * It will retry the command up to maxAttempts times
  * TODO: Add batching to prevent bottlenecks when executing in a huge cluster.
  */
-func (executor *GPDBExecutor) ExecuteClusterCommand(scope Scope, commandList []ShellCommand) *RemoteOutput {
+func (executor *GPDBExecutor) ExecuteClusterCommandWithRetries(scope Scope, commandList []ShellCommand, maxAttempts int, retrySleep time.Duration) *RemoteOutput {
 	length := len(commandList)
 	finished := make(chan int)
 	numErrors := 0
 	for i := range commandList {
 		go func(index int) {
+			var (
+				out    []byte
+				err    error
+				stderr bytes.Buffer
+			)
 			command := commandList[index]
-			var stderr bytes.Buffer
-			cmd := command.Command
-			cmd.Stderr = &stderr
-			out, err := cmd.Output()
+			for attempt := 1; attempt <= maxAttempts; attempt++ {
+				stderr.Reset()
```

Comment:
Do we also want to flush stdout if there were multiple attempts?

Reply:
The stdout gets refreshed when we create the new command with resetCmd